### DISCUSSION TITLE
Which type is correct?

### DISCUSSION THREAD

**Discussion by karaka200** [symfony/symfony#44687] [Q&A] [answered]
Which is correct?
```
#[ORM\Column]
private int $id;

#[ORM\Column(length: 180, unique: true)]
private string $username;
```

or

```
#[ORM\Column(type: 'integer')]
private $id;

#[ORM\Column(type: 'string', length: 180, unique: true)]
private $username;
```

PHPStorm say the first is true!

**Comment by stof** [comment, marked as answer]
First, you are asking in the wrong place, as you ask support about https://github.com/doctrine/orm, not about Symfony.

In doctrine/orm 2.10, both are correct as the `type` of the column defaults to being guessed based on the property type.
However, be careful when using typed properties if you use your entity as a form model directly. Form models can contain invalid data (that's why the validator is run on them as part of the submission).

**Reply by karaka200** [reply]
Thank you. In that case, which one do you recommend for more security?

**Reply by karaka200** [reply]
Some places use this:

```
#[ORM\Column(type: 'integer')]
private int $id;

#[ORM\Column(type: 'string', length: 180, unique: true)]
private string $username;
```
Is it better than others?!!

**Reply by karaka200** [reply]
And see @weaverryan issue:
https://github.com/symfony/maker-bundle/issues/982
